src/modules/menu/components/timeslider/TimeSlider.vue

```diff
@@ -0,0 +1,468 @@
+                } else if (deltaX > 0) {
+                    if (futureYearIndex > absoluteDeltaIndex) {
+                        futureYearIndex -= absoluteDeltaIndex
+                    } else if (futureYearIndex > 0) {
+                        futureYearIndex--
+                    }
+                }
+                const futureYear = ALL_YEARS[futureYearIndex]
+                // checking that this is a valid year in the context of currently displayed data
+                if (this.yearsWithData.includes(futureYear)) {
+                    // reset of the starting position for delta calculation
+                    this.cursorX = currentPosition
+                    this.currentYear = futureYear
+                }
+            }
+        },
+        releaseCursor() {
+            this.yearCursorIsGrabbed = false
+            window.removeEventListener('mousemove', this.listenToMouseMove)
+            window.removeEventListener('touchmove', this.listenToMouseMove)
+            window.removeEventListener('mouseup', this.releaseCursor)
+            window.removeEventListener('touchend', this.releaseCursor)
+            this.setPreviewYear(this.currentYear)
+        },
+        togglePlayYearsWithData() {
+            this.playYearsWithData = !this.playYearsWithData
+            if (this.playYearsWithData) {
+                // if current year is the last (most recent) one, we set the starting year for our player to the oldest
+                if (this.currentYear === this.yearsWithData[0]) {
+                    this.setCurrentYearAndDispatchToStore(this.yearsWithData.slice(-1)[0])
+                }
+                this.playYearInterval = setInterval(() => {
+                    const currentYearIndex = this.yearsWithData.indexOf(this.currentYear)
+                    // if last (most recent) year, we stop the player
+                    if (currentYearIndex === 0) {
+                        clearInterval(this.playYearInterval)
+                        this.playYearInterval = null
+                        this.playYearsWithData = false
+                    } else {
+                        this.setCurrentYearAndDispatchToStore(
+                            this.yearsWithData[currentYearIndex - 1]
+                        )
+                    }
+                }, 1000)
+            } else {
+                clearInterval(this.playYearInterval)
+                this.playYearInterval = null
+            }
+        },
+    },
+}
+</script>
+<style lang="scss">
+@import '../../../../scss/webmapviewer-bootstrap-theme';
+
+.time-slider {
+    background: rgba(255, 255, 255, 0.9) !important;
+    &-bar {
+        &-cursor {
+            $cursor-height: 34px;
+            position: absolute;
+            top: 0.75 * $spacer;
+            height: $cursor-height;
+            width: 92px;
+            &-grip {
+                cursor: grab;
+                &.grabbed {
+                    cursor: grabbing;
+                }
```

Comment:
The `grabbing` cursor should be set on all viewport object once grabbed, because otherwise when sliding quickly or when sliding and moving the mouse above the year button, the cursor changes back to normal.